engine.py

```diff
@@ -801,10 +803,18 @@ def convert_textures(self, temp_texture_list):
 
         files_converted = []
         texture_list = []
+        futures = []
 
         if not temp_texture_list:
             return
 
+        if self.rm.threads > 0:
+            num_threads = self.rm.threads
+        else:
+            num_threads = cpu_count() + self.rm.threads
+
+        pool = ThreadPoolExecutor(num_threads)
```

Comment:
So this looks good.  Actually I'm adding some of the concurrent.futures stuff for an upcoming display driver built in to blender.  

But I think the number of slots in the executor (and the options to txmake) maybe should be tweaked a bit.  txmake is internally multithreaded.  It's just you are seeing bad usage of it.  There is a -t:x option to txmake already which defaults to number of cpu_count().  Also on a tangent does cpu_count return the number of physical cores or hyperthread cores?  

My point being that if you have 8 cores and 8 textures to txmake and each txmake would independently use ~25% of your total cpu you might be oversubscribing a bit and end up being slower overall although one could argue that's up to the OS to manage.  It might be worth testing if overall the txmake time is faster if you limit the number of slots in the Executor or if you limit the thread usage of txmake via -t:N where N is some value < cpu_count